### PR TITLE
Support cross platform resource manager without WCF support

### PR DESCRIPTION
**Pull request by Toxantron**
I thought it would be cool, if we could use the resource manager x-platform, if we excluded WCF. Maybe with a preconfigured database for use cases like minimal IoT applications.

### CODE REVIEW

File: src/Moryx.Resources.Management/Resources/ResourceGraph.cs [hunk 1]
```diff
@@ -24,54 +25,65 @@ internal class ResourceGraph : IManagedResourceGraph
         /// </summary>
         public IResourceTypeController TypeController { get; set; }
 
+        private ReaderWriterLockSlim _graphLock = new ReaderWriterLockSlim();
```
Comment (dbeuchler) [inline src/Moryx.Resources.Management/Resources/ResourceGraph.cs:28]: Can be readonly

File: src/Moryx.Resources.Management/Moryx.Resources.Management.csproj [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <Description>ResourceManagement module composing and maintaining the resource graph as the habitat for digital twins of manufacturing assets.</Description>
     <CreatePackage>true</CreatePackage>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net45'">
+    <DefineConstants>HAVE_WCF</DefineConstants>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net45'">
```
Comment (dbeuchler) [inline src/Moryx.Resources.Management/Moryx.Resources.Management.csproj:14]: Two times the same property group condition. Just seperate the constants (array) by semicolon;

File: src/Moryx.Resources.Management/ModuleController/ModuleController.cs [hunk 3]
```diff
@@ -105,16 +115,16 @@ protected override void OnStop()
             resourceManager.Stop();
         }
 
-        #endregion
+#endregion
 
-        #region FacadeContainer
+#region FacadeContainer
```
Comment (dbeuchler) [inline src/Moryx.Resources.Management/ModuleController/ModuleController.cs:120]: The default resharper setting is that the region definition is on the same level with the first line of code of the region. Compiler definitions (like if, else...) are starting at the beginning.

**Review by dbeuchler — CHANGES_REQUESTED** [review]

File: src/Moryx.Resources.Management/ModuleController/ModuleController.cs (hunk 3 above)
Comment (Toxantron) [reply src/Moryx.Resources.Management/ModuleController/ModuleController.cs:120]: Sorry, Studio or ReSharper AutoFormatted that when inserting the Region.

Comment (Toxantron) [reply src/Moryx.Resources.Management/ModuleController/ModuleController.cs:120]: I took a second look. That whole region is overblown. The class has 112 LOC.

**Review by dbeuchler — APPROVED** [review]

**Pull request merged**